extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -525,12 +533,20 @@ UnremovableBeanBuildItem unremovableRoutesBuilders() {
         }
 
         @Overridable
-        @Record(value = ExecutionTime.RUNTIME_INIT, optional = true)
+        @Record(value = ExecutionTime.STATIC_INIT, optional = true)
         @BuildStep(onlyIf = Flags.MainEnabled.class)
         CamelReactiveExecutorBuildItem reactiveExecutor(CamelMainRecorder recorder) {
             return new CamelReactiveExecutorBuildItem(recorder.createReactiveExecutor());
         }
 
+        @Overridable
+        @Record(value = ExecutionTime.RUNTIME_INIT, optional = true)
+        @BuildStep(onlyIf = Flags.MainEnabled.class)
+        CamelInitializedReactiveExecutorBuildItem initReactiveExecutor(
+                CamelMainRecorder recorder, CamelReactiveExecutorBuildItem executor) {
```

Comment:
Are the `recorder` param and `@Record` annotation required? If so, could you please add a comment explaining why?

Reply:
Another comment I'd appreciate here is an explanation why is this optional/overridable, something like

```
This produces the default CamelInitializedReactiveExecutorBuildItem simply by repackaging the excutor produced by reactiveExecutor(). Both this method and reactiveExecutor() are overridable to allow plugging in a Vertx based executor by adding the camel-quarkus-reactive-executor extension to the class path.
```